pkg/controller/factory/factory.go

```diff
@@ -34,6 +36,11 @@ type namespaceInformer struct {
 	namespaces sets.String
 }
 
+// PostRunHook specify a function that will run after controller is ran.
+// The context is cancelled when the controller is asked to shutdown and the post start hook should terminate as well.
+// The syncContext allow access to controller queue and event recorder.
+type PostRunHook func(ctx context.Context, syncContext SyncContext) error
```

Comment:
ERRTOOMANYCONTEXTS

Reply:
upstream calls SyncContext `ReconcileRequest`... I don't think it is worth renaming it now...